packages/core/src/analytics/dispatch.ts

```diff
@@ -0,0 +1,51 @@
+import { CoreContext } from '../context'
+import { CoreSegmentEvent, Callback } from '../events/interfaces'
+import { EventQueue } from '../queue/event-queue'
+import { isOffline } from '../connection'
+import { Emitter } from '../emitter'
+import { invokeCallback } from '../callback'
+
+type DispatchOptions = {
+  timeout?: number
+  debug?: boolean
+  callback?: Callback
+  retryQueue?: boolean
+}
+
+export async function dispatch(
+  event: CoreSegmentEvent,
+  queue: EventQueue,
+  emitter: Emitter,
+  options?: DispatchOptions
+): Promise<CoreContext> {
+  const ctx = new CoreContext(event)
+  emitter.emit('message_dispatch_pending', ctx) // TODO: inspectorHost.triggered?.(ctx as any)
+
+  if (isOffline() && !options?.retryQueue) {
+    return ctx
+  }
+
+  const startTime = Date.now()
+  let dispatched: CoreContext
+  if (queue.isEmpty()) {
+    dispatched = await queue.dispatchSingle(ctx)
+  } else {
+    dispatched = await queue.dispatch(ctx)
+  }
+  const elapsedTime = Date.now() - startTime
+  const timeoutInMs = options?.timeout
+
```

Comment:
Note: This might be a good spot to remove the context from the priorityQueue's `seen` map. That can be done in a follow-up PR (I can create an issue for that unless you already fixed that memory leak) - basically want to make sure we don't let that map just grow especially in node.js.

Reply:
I didn't fix the memory leak, but that's a good callout -- we can prune old events.